streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -274,18 +274,202 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
       case DeregisterReceiver(streamId, message, error) =>
         deregisterReceiver(streamId, message, error)
         context.reply(true)
+      // Local messages
+      case AllReceiverIds =>
+        context.reply(receiverTrackingInfos.keys.toSeq)
       case StopAllReceivers =>
         assert(isTrackerStopping || isTrackerStopped)
         stopReceivers()
         context.reply(true)
     }
 
+    private def startReceiver(receiver: Receiver[_]): Unit = {
+      val checkpointDirOption = Option(ssc.checkpointDir)
+      val serializableHadoopConf =
+        new SerializableConfiguration(ssc.sparkContext.hadoopConfiguration)
+
+      // Function to start the receiver on the worker node
+      val startReceiverFunc = new StartReceiverFunc(checkpointDirOption, serializableHadoopConf)
+
+      receiverPreferredLocations(receiver.streamId) = receiver.preferredLocation
+      val receiverId = receiver.streamId
+
+      if (isTrackerStopping() || isTrackerStopped()) {
+        onReceiverJobFinish(receiverId)
+        return
+      }
+
+      val scheduledLocations = schedulingPolicy.scheduleReceiver(
+        receiverId,
+        receiver.preferredLocation,
+        receiverTrackingInfos,
+        getExecutors(ssc))
+      updateReceiverScheduledLocations(receiver.streamId, scheduledLocations)
+
+      // Create the RDD using the scheduledLocations to run the receiver in a Spark job
+      val receiverRDD: RDD[Receiver[_]] =
+        if (scheduledLocations.isEmpty) {
+          ssc.sc.makeRDD(Seq(receiver), 1)
```

Comment:
nit: Add name to the RDD so that it appears in the UI with useful names like "Receiver 1"